lib/stream.js

```diff
@@ -1,11 +1,14 @@
 'use strict';
 
-module.exports = Stream;
+var Stream = require('internal/streams/stream').Stream;
+
+// wrap the old-style stream
+require('internal/streams/legacy')(Stream);
```

Comment:
You couldn't just move those files because that would break someone trying to `require()` them? Unless you mean leave behind similarly-named files that simple re-export?

Also, I think the point of this PR was to separate the actual old `Stream` implementation from the other things, like the re-exporting of the various stream types.